app/services/db/private-cloud-costs.ts

```diff
@@ -279,13 +279,27 @@ export async function getYearlyCosts(licencePlate: string, yearString: string) {
   return { items };
 }
 
+async function computeTotalCost(allProducts: { licencePlate: string }[], year: number, month: number): Promise<number> {
+  let totalCost = 0;
+
+  for (const { licencePlate } of allProducts) {
+    const { currentTotal, grandTotal } = await getMonthlyCosts(licencePlate, year, month);
+    const cost = grandTotal > -1 ? grandTotal : currentTotal;
+    totalCost += cost;
+  }
+
+  return totalCost;
+}
+
 export async function getAdminMonthlyCosts(year: number, oneIndexedMonth: number) {
-  const products = await prisma.privateCloudProduct.findMany({
-    select: {
-      name: true,
-      licencePlate: true,
-    },
-  });
+  const [products, totalCount] = await Promise.all([
+    prisma.privateCloudProduct.findMany({
+      select: { name: true, licencePlate: true },
+    }),
+    prisma.privateCloudProduct.count(),
+  ]);
+
+  const totalCost = await computeTotalCost(products, year, oneIndexedMonth);
```

Comment:
please check again what is inside `computeTotalCost`; it doubles the calculation process